doc/source/reference/stats.rst

```diff
@@ -7,4 +7,5 @@
    :hidden:
 
    stats._result_classes
+   stats._warnings_errors
```

Comment:
I think the intent was _not_ to show these publicly. Even in `main`, "Result Classes" shows up, so this is not new. Can we get rid of these but still link to e.g. `BinomTestResult` documentation within the `binomtest` documentation?

![image](https://user-images.githubusercontent.com/6570539/161403197-49a39393-ed95-47af-855d-65dbe95bac3f.png)